tsl/src/nodes/decompress_chunk/decompress_chunk.c

```diff
@@ -1839,11 +1841,31 @@ create_compressed_scan_paths(PlannerInfo *root, RelOptInfo *compressed_rel, Comp
 		 * decompression
 		 */
 		List *orig_pathkeys = root->query_pathkeys;
+		List *orig_eq_classes = root->eq_classes;
 		build_compressed_scan_pathkeys(sort_info, root, root->query_pathkeys, info);
 		root->query_pathkeys = sort_info->required_compressed_pathkeys;
+
+		/* We can optimize iterating over EquivalenceClasses by reducing them to
+		 * the subset which are from the compressed chunk. This only works if we don't
+		 * have joins involved since those use eclass_indexes which is not valid
```

Comment:
Can we also temporarily reset the eclass_indexes to null, to make this more consistent?